app/lib/three_scale/json_validator.rb

```diff
@@ -35,7 +35,7 @@ def initialize(json, schema_reader_options = {})
 
     def fully_validate(schema, opts = {})
       options = opts.reverse_merge(schema_reader: schema_reader)
-      ::JSON::Validator.fully_validate(schema, json, options)
+      self.class.fully_validate(schema, json, options)
```

Comment:
Honestly I have no idea how could the `fully validate` test ever work previously. But it worked when run after the `autoload schema files` test. If you have any idea how was this possible, let me know as it is super strange for me and I couldn't make any sense out of it except for some subtle mocha quirk.

Reply:
But this change is not equivalent to what we had before?

Reply:
I don't find it a good design to inherit from `ClassA` and then call into `ClassA::method`. May easily lead to a surprising behavior.

 It's definitely not universally equivalent. But I think it is functionally equivalent in this case.

Reply:
Actually changing this breaks validation. So I resorted to updating the test case. See my comment there for a rationale https://github.com/3scale/porta/pull/3618/commits/41e532688cc026e1a829d3259670689eb7ea47a2#r1393223254

Reply:
Also based on this, I will attempt to update mocha to 2.1.0 afterwards.